src/LOAuth.ts

```diff
@@ -93,6 +96,31 @@ class LOAuth {
     }
   }
 
+  private _getStateForClientOAuth(options: LOAuth.Config) {
+    this._decodeAppState();
+
+    let state = JSON.stringify({
```

Comment:
It looks like state is completely overwritten if `queryParameters.get('state')` is true. To avoid the `let` and extract work, should this stringify be moved into an else block instead?

Reply:
@mdlavin I don't think that would be any clearer.  The state is overwritten, but the initial stringified state is needed to build the new value in the conditional:
```state = encodeURIComponent(state)```
So if I moved to an if/else block I would need to have the ```JSON.stringify``` call in both places

Reply:
Oh, right. Sorry about that.